x-pack/filebeat/module/ibmmq/errorlog/ingest/pipeline.json

```diff
@@ -0,0 +1,69 @@
+{
+    "description": "Pipeline for parsing MQ error logs.",
+    "processors": [
+        {
+            "gsub": {
+                "field": "message",
+                "pattern": "^[\\-]{5}[a-z0-9\\. :]*[\\-]{5,}",
+                "replacement": ""
+            }
+        },
+        {
+            "gsub": {
+                "field": "message",
+                "pattern": "\n",
+                "replacement": " "
+            }
+        },
+        {
+            "gsub": {
+                "field": "message",
+                "pattern": "[ ]{2,}",
+                "replacement": " "
+            }
+        },
+        {
+            "trim": {
+                "field": "message"
+            }
+        },
+        {
+            "rename": {
+                "field": "@timestamp",
+                "target_field": "event.created"
+            }
+        },
+        {
+            "grok": {
+                "field": "message",
+                "patterns": [
+                    "^%{DATA:log_timestamp} -"
+                ]
+            }
+        },
+        {
+            "grok": {
+                "field": "message",
+                "patterns": [
+                    "Process\\(%{DATA:process.pid}\\) User\\(%{WORD:user.name}\\) Program\\(%{DATA:process.title}\\) Host\\(%{DATA:host.hostname}\\) Installation\\(%{WORD:ibmmq.errorlog.installation}\\) VRMF\\(%{DATA:service.version}\\)( QMgr\\(%{DATA:ibmmq.errorlog.qmgr}\\))?( Time\\(%{TIMESTAMP_ISO8601:@timestamp}\\))?( RemoteHost\\(%{DATA:destination.address}\\))?( ArithInsert1\\(%{DATA:ibmmq.errorlog.arithinsert1}\\))?( ArithInsert2\\(%{DATA:ibmmq.errorlog.arithinsert2}\\))?( CommentInsert1\\(%{DATA:ibmmq.errorlog.commentinsert1}\\))?( CommentInsert2\\(%{DATA:ibmmq.errorlog.commentinsert2}\\))?( CommentInsert3\\(%{DATA:ibmmq.errorlog.commentinsert3}\\))? (?=AMQ[0-9]{4})%{DATA:error.id}((?<=AMQ[0-9]{4}[A-Z])%{DATA:log.level})?: %{DATA:ibmmq.errorlog.errordescription} EXPLANATION: %{DATA:ibmmq.errorlog.explanation} ACTION: %{DATA:ibmmq.errorlog.action}$"
+                ]
+            }
+        },
+        {
+            "remove": {
+                "field": [
+                    "log_timestamp",
+                    "message"
```

Comment:
Note for myself: Should we really fully remove message or keep at least something inside?